ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -311,24 +311,50 @@ module.exports = function CesiumMap(
       map.scene.camera.moveEnd.addEventListener(callback)
     },
     doPanZoom: function(coords) {
-      const cartArray = coords.map(function(coord) {
-        return Cesium.Cartographic.fromDegrees(
-          coord[0],
-          coord[1],
-          map.camera._positionCartographic.height
-        )
-      })
-      if (cartArray.length === 1) {
-        const point = Cesium.Ellipsoid.WGS84.cartographicToCartesian(
-          cartArray[0]
-        )
-        this.panToCoordinate(point, 2.0)
-      } else {
-        const rectangle = Cesium.Rectangle.fromCartographicArray(cartArray)
+      if (Array.isArray(coords[0][0])) {
```

Comment:
✏️ 	I think this logic can be made more readable by extracting a `getLargestBbox` function (or something similar) and possibly a `isMultiPolygon` function for this conditional